### DISCUSSION TITLE
обновление

### DISCUSSION THREAD

**Discussion by yurikfm** [petrovvlad/freeLib#143] [General]
Всем привет.
Как не перекачивая папку freelib, его обновить с последующей переустановкой самой программы?
Система линукс.

**Comment by Sreg-Ko** [comment]
Вы хотя бы уточните что конкретно вы имели ввиду?

**Comment by petrovvlad** [comment]
Не понял вопрос. Вы хотите обновить программу?  Программа обновляется установкой новой версии.

**Comment by yurikfm** [comment]
Да, немножко не ясно выразился.
Я слежу за этим продуктом и вижу изменения. После того как сгенерирована папка freelib(git clone --recurse-submodules https://github.com/petrovvlad/freeLib.git) мы же устанавливаем саму программу Freelib.
1. Папку freelib(sources) на ПК как можно обновить? Не удаляя устаревшую версию.
2. Переустановка программы Freelib
П.С. Пока я просто удаляю папку freelib(sources) из ПК и удаляю саму программу Freelib.
Скачивая по новой, повторяю все те шаги по установке программы. Согласитесь, это неправильно ...

**Reply by Sreg-Ko** [reply]
Если собираете из исходников, переустановка наверноее единственный вариант.

**Comment by Sreg-Ko** [comment]
После того как вы скачаете git clone --recurse-submodules https://github.com/petrovvlad/freeLib.git
Вы делаете сборку mkdir freeLib/build && cd freeLib/build
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD:STRING=20 -DCMAKE_INSTALL_PREFIX=/usr .. && cmake --build . -j4
Потом делаете установку sudo make install
После установки у вас в папке build появится файл install_manifest.txt
В этом файле указано в какие папки и что установлено.
Если не хотите удалять и по новому устанавливать freelib, просто замените старый файл freelib новым из папки /build/freeLib/src

**Reply by Sreg-Ko** [reply]
Конечно это не есть хорошо (лучше всетаки удалить программу), но как вариант.

**Comment by yurikfm** [comment]
Ну думаю саму программу можно и удалять\устанавливать.
А вот папку freelib(sources) как то можно обновлять без удаления самой папки? По типу синхронизации файлов, старые заменяются 
новыми.
Конечно потом будем ставить поновой из той папки, но уже новую версию Freelib. Исправленную так сказать ...

**Reply by Sreg-Ko** [reply]
Ну так все равно компилить по новому придется. Да и если что то еще изменится кроме sources. Честно не вижу смысла тягать с github по частям. 
Я для себя вообще сделал скрипт для установки.

#!/bin/sh
cd ~
rm  -rf ~/freelib
git clone --recurse-submodules https://github.com/petrovvlad/freeLib.git
mkdir freeLib/build && cd freeLib/build
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD:STRING=20 -DCMAKE_INSTALL_PREFIX=/usr .. && cmake --build . -j20
sudo cmake --install . --config Release
cd ~

**Reply by Sreg-Ko** [reply]
И такой же скрипт для удаления(запускать под sudo)
#!/bin/sh
rm -r /usr/bin/kindlegen
rm -rf /usr/share/freelib
rm -r /usr/bin/freelib
rm -r /usr/share/applications/freelib.desktop
rm -r /usr/share/icons/hicolor/256x256/apps/freelib.png
rm -r /usr/share/icons/hicolor/48x48/apps/freelib.png
rm -r /usr/share/icons/hicolor/scalable/apps/freelib.svg
rm -r /usr/share/icons/hicolor/16x16/apps/freelib.png
rm -r /usr/share/icons/hicolor/128x128/apps/freelib.png
rm -r /usr/share/metainfo/freelib.appdata.xml

**Reply by yurikfm** [reply]
Ну я готов поновой компилить, благо исходники не каждые 5 минут редактируются ...
В общем нужно клонированную папку git clone --recurse-submodules https://github.com/petrovvlad/freeLib.git периодически обновлять.
По мере исправлений ошибок.

**Reply by Sreg-Ko** [reply]
ну тут мне кажется без вариантов).
